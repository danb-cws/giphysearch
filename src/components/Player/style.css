.Player {
  color: var(--primary-color-inverse);
  background-color: var(--secondary-bg-color);
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 0;
  min-width: 0;
}
.Player--img {
  object-fit: contain;
  height: auto;
  width: 100%;
  max-height: 100%;
}
.Player--errortext {
  display: flex;
  justify-content: center;
  align-items: center;
  color: var(--secondary-color-inverse);
  background-color: var(--secondary-bg-color-inverse);
  font-size: 2rem;
  z-index: 2;
  padding: 1rem;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}
.Player--loadingtext {
  color: var(--secondary-color-inverse);
}
