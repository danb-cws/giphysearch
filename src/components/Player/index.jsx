import React, { Component } from "react";
import "./style.css";
import Spinner from "../Spinner/";
import ShareTool from "../ShareTool/";
import * as config from "config";

class Player extends Component {
  state = { localCurrId: undefined, mainGif: {}, imageIsLoaded: undefined };

  fetchPlayerImage = () => {
    fetch(
      `${config.GIPHY_ENDPOINT}${this.props.currentId}?&api_key=${
        config.GIPHY_API_KEY
      }`
    )
      .then(response => {
        if (response.ok) {
          return response.json();
        } else {
          throw new Error(response.statusText);
        }
      })
      .then(json => {
        this.setState({
          localCurrId: this.props.currentId,
          mainGif: json.data,
          imageIsLoaded: false
        });
      })
      .catch(() => console.error);
  };

  isEmpty = obj => {
    for (var key in obj) {
      if (obj.hasOwnProperty(key)) return false;
    }
    return true;
  };

  hideSpinner = () => {
    this.setState({
      imageIsLoaded: true
    });
  };

  componentDidUpdate = () => {
    if (
      this.props.currentId !== 0 && // ie. not 'no term' or 'no results'
      this.props.currentId !== this.state.localCurrId // prevent the fetch on every render if id unchanged
    ) {
      this.fetchPlayerImage(this.props.currentId);
    }
  };

  render() {
    const { mainGif, imageIsLoaded } = this.state;
    if (this.isEmpty(mainGif)) {
      return <div className="Player">Loading json...</div>;
    }
    return (
      <div className="Player">
        <Spinner showSpinner={!imageIsLoaded} />
        <ShareTool />
        <img
          className="Player--img"
          src={mainGif.images.downsized.url}
          alt={mainGif.title}
          onLoad={this.hideSpinner}
        />
      </div>
    );
  }
}

export default Player;
