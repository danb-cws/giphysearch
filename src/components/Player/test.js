import React from "react";
import { shallow } from "enzyme";
import Player from "./";
import testjson from "mock_json/gifMock.json";

const el = shallow(<Player currentChoiceData={testjson} />);

describe("Player tests", () => {
  it("Renders the Player component", () => {
    expect(el.exists()).toBe(true);
  });
  it("Matches snapshot", () => {
    expect(el).toMatchSnapshot();
  });
});
