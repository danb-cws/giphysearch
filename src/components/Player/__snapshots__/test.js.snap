// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player tests Matches snapshot 1`] = `
<section
  className="Player"
>
  <Spinner
    showSpinner={true}
  />
  Loading json...
</section>
`;
